apps/root/src/pages/transfer/transfer-container/components/transfer-button/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { parseUnits } from '@ethersproject/units';
+import useContactListService from '@hooks/useContactListService';
+import { useTransferState } from '@state/transfer/hooks';
+import { FormattedMessage } from 'react-intl';
+import { Button, Typography } from 'ui-library';
+import { PROTOCOL_TOKEN_ADDRESS } from '@common/mocks/tokens';
+import { Token, TokenType } from '@types';
+import useActiveWallet from '@hooks/useActiveWallet';
+
+const TransferButton = () => {
+  const { token, amount, recipient } = useTransferState();
+  const contactListService = useContactListService();
+  const activeWallet = useActiveWallet();
+
+  const parsedAmount = parseUnits(amount || '0', token?.decimals);
+
+  const onTransfer = async () => {
+    if (!recipient || !token || parsedAmount.lte(0) || !activeWallet) {
+      return;
+    }
+    try {
+      const isProtocolToken = token.address === PROTOCOL_TOKEN_ADDRESS;
+      const parsedToken: Token = { ...token, type: isProtocolToken ? TokenType.BASE : TokenType.ERC20_TOKEN };
+
+      await contactListService.transferTokenToContact({
```

Comment:
We should use the transaction modal here for error and loading, no need for success if we are going to show the TransactionConfirmation

Reply:
We also need to add this to the transactions that we track for the user and build messages for each of the states, check `apps/root/src/hooks/useBuildTransactionDetail.ts` and the other `useBuildTransaction*`